ansible/roles/debops.postldap/tasks/main.yml

```diff
@@ -0,0 +1,41 @@
+---
+
+- name: Make sure required POSIX group exist
+  group:
+    name: '{{ postldap__vmail_posix_group |
+              d(postldap__vmail_posix_user) }}'
+    state: 'present'
+    system: True
+
+- name: Make sure required POSIX system account exist
+  user:
+    name: '{{ postldap__vmail_posix_user }}'
+    state: 'present'
+    comment: 'Postfix Virtual Mail user'
+    group: '{{ postldap__vmail_posix_group |
+               d(postldap__vmail_posix_user) }}'
+    home: '{{ postldap__mailbox_base }}'
+    create_home: True
+    system: True
+    shell: '/usr/sbin/nologin'
+    skeleton: null
+
+- name: "Make sure vmail home directory ({{ postldap__mailbox_base }}) exists with restrictive permissions"
+  file:
+    dest: '{{ postldap__mailbox_base }}'
+    state: 'directory'
+    owner: '{{ postldap__vmail_posix_user }}'
+    group: '{{ postldap__vmail_posix_group }}'
+    mode: '0750'
+
+- name: Get vmail group info
+  getent:
+    database: 'group'
+    split: ':'
+    key: '{{ postldap__vmail_posix_group }}'
+
+- name: Get vmail passwd info
+  getent:
+    database: 'passwd'
+    split: ':'
+    key: '{{ postldap__vmail_posix_user }}'
```

Comment:
I think that having the UID/GID gathered at runtime might not be idempotent, if the playbook runs only with the `role::postfix` tag, but I noticed that you added it to the `debops.postldap` role as well, which should cover this situation. Still, I think that getting the UID/GID via an Ansible local fact and using the facts to define them in the default variables instead of `getent_group` and `getent_passwd` variables might be a bit better approach.

And the role should include the fact script anyway, to indicate that it has been applied on the host. It could be just a basic script without the UID/GID gathering if you don't want to implement it at this point, that can come later.

Reply:
> I think that having the UID/GID gathered at runtime might not be idempotent, 

you do not want idempotency here, you want the correct values at runtime.

> Still, I think that getting the UID/GID via an Ansible local fact and using the facts to define them in the default variables instead of `getent_group` and `getent_passwd` variables might be a bit better approach.

One second after writing the UID/GID values in the fact script, this values may be old and have changed. There is no guarantee they will be kept up to date and do not change. Thus getting the values just when you need them from the official POSIX/Linux user store is a good way to have up-to-date data.

> And the role should include the fact script anyway, to indicate that it has been applied on the host. It could be just a basic script without the UID/GID gathering 

What do you mean, just a simple fact script, so "we" know debops.postldap was executed before?

Reply:
> One second after writing the UID/GID values in the fact script, this values may be old and have changed. There is no guarantee they will be kept up to date and do not change. Thus getting the values just when you need them from the official POSIX/Linux user store is a good way to have up-to-date data.

I could say the same about the task-based method. After all, each task is in reality a separate Python script that gets sent to the remote host for execution, and returns with the response. If we are talking about second-based resolution, depending on the network and I/O latency the UID/GID could be changed even between first and second `getent` task.

> > And the role should include the fact script anyway, to indicate that it has been applied on the host. It could be just a basic script without the UID/GID gathering
> 
> What do you mean, just a simple fact script, so "we" know debops.postldap was executed before?

Yes, that's one of the reasons. The other is that DebOps roles are purposefully kept as separate as possible, and Ansible facts guarantee loose coupling between various roles. Not to mention that other roles like `debops.dovecot` could use the `postldap.fact` script facts for their own purposes, for example to configure UID/GID access if it's ever needed somewhere. In that case there will be no need to create another set of two `getent` tasks, because the facts are available to all roles.

Reply:
So, added a facts script with the data from user module.
Does this works for you?